src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpSimplifyTypeNamesDiagnosticAnalyzer.cs

```diff
@@ -47,14 +49,16 @@ SyntaxKind.DelegateDeclaration or
                 SyntaxKind.EnumDeclaration;
         }
 
-        protected override ImmutableArray<Diagnostic> AnalyzeCodeBlock(CodeBlockAnalysisContext context)
+        protected override ImmutableArray<Diagnostic> AnalyzeCodeBlock(CodeBlockAnalysisContext context, SyntaxNode root)
         {
+            Debug.Assert(context.CodeBlock.DescendantNodesAndSelf().Contains(root));
+
             var semanticModel = context.SemanticModel;
             var cancellationToken = context.CancellationToken;
 
             var options = context.GetCSharpAnalyzerOptions().GetSimplifierOptions();
             using var simplifier = new TypeSyntaxSimplifierWalker(this, semanticModel, options, ignoredSpans: null, cancellationToken);
-            simplifier.Visit(context.CodeBlock);
+            simplifier.Visit(root);
```

Comment:
can we get this anlayzer to not analyze blocks unrelated to the filter span?  (if this was already done in this PR, great).